runtime-sdk/modules/evm/src/backend.rs

```diff
@@ -277,7 +277,9 @@ impl<'c, C: Context, Cfg: Config> ApplyBackendResult for Backend<'c, C, Cfg> {
                         let val: H256 = value.into();
 
                         let ctx = self.ctx.get_mut();
-                        if value == primitive_types::H256::default() {
+                        if value == primitive_types::H256::default()
+                            && (Cfg::CONFIDENTIAL && !Cfg::CONFIDENTIAL_CONSTIFY)
```

Comment:
Shouldn't this be:
```suggestion
                            && (!Cfg::CONFIDENTIAL || !Cfg::CONFIDENTIAL_CONSTIFY)
```

Also do I see this right that if `CONFIDENTIAL_CONSTIFY` is enabled, there is no way to remove stuff from storage? This doesn't seem like the best thing.

Reply:
This is a bit aggressive in terms of constifying, I agree. There are a few levels of constification. I think the big ones are 1) time/gas, 2) io data, 3) io patterns. `CONFIDENTIAL_CONSTIFY` does 1 and 2 currently. Maybe this should be an enum

Reply:
Actually, I'm just going to remove it. I'm not going to enable it, so it's just more code to maintain. For future readers, I'd do something like this

```rust
/// How much side-channel resistance do you want the runtime to have?
///
/// Mitigations can be applied by the application, but then the app authors
/// need to know about the issue and then write secure code (difficult).
/// Instead the runtime can apply mitigations, but this increases costs and
/// introduces incompatibilities for all apps including ones that don't need
/// side-channel resistance.
#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
#[non_exhaustive]
enum Constancy {
    /// Operations may be data dependent. This is the default option.
    #[default]
    None,
    /// Data dependencies that are easy/cheap to mitigate are removed.
    ///
    /// Specifically: tx calldata zero/non-zero bytes cost the same, and
    /// binary opcodes do not short circuit.
    ///
    /// Caveats: nothing else is constified
    Light,
    /// Expensive mitigations are also applied.
    ///
    /// Specifically: storage set and reset gas become the same, reset gas is not
    /// refunded, and storage values are not removed when reset to zero.
    ///
    /// Caveats: Storage observers may still infer data based on access patterns.
    /// Short-circuiting operators within app code still need to be made constant-time.
    Heavy,
    /// Side channels are eliminated. Applications do not need to write constant-time code.
    ///
    /// Caveat: super expensive, even when you don't need confidentiality
    // TODO: oram, differentially private time+gas
    Full,
}
```